File: ScriptLanguage/Lab1/lab1.js
const { performance } = require('perf_hooks');

const arr = Array.from({ length: 10000 }, () => Math.floor(Math.random() * 100000) + 1);

function myFunc(arr) {
    let count = 0;
    const len = arr.length;
    for (let i = 0; i < len; i++) {
        for (let j = i + 1; j < len; j++) {
            if (arr[i] === arr[j]) {
                count++;
            }
        }
    }
    return count;
}

const start = performance.now();
const c = myFunc(arr);
const finish = (performance.now() - start); 

console.log(`Количество дубликатов в массиве ${c}`);
console.log(`Время работы алгоритма на JS: ${finish.toFixed(2)} миллисекунд`);

File: ScriptLanguage/Lab1/lab1.py
# Алгоритм поиск количества дубликатов в одномерном массиве
import time
import random

arr = [random.randint(1, 100000) for _ in range(10000)]
def my_func(arr):
    count = 0
    for i in range(len(arr)):
        for j in range(i+1, len(arr)):
            if arr[i] == arr[j]:
                count+=1
    return count
start = time.time()
c = my_func(arr)
finish = (time.time() - start) * 1000
print(f"Количество дубликатов в массиве {c}")
print(f"Время работы алгоритма на Python: {finish:.2f} миллисекунд")